tests/bdd/features/volume/resize/test_resize_offline.py

```diff
@@ -0,0 +1,448 @@
+    all_the_replicas_of_the_volume_should_be_resized_to_the_new_capacity()
+
+
+@then("the onlined replica should be rebuilt")
+def the_onlined_replica_should_be_rebuilt():
+    """the onlined replica should be rebuilt."""
+    vol = ApiClient.volumes_api().get_volume(VOLUME_UUID)
+    target = vol.state.target
+    childlist = target["children"]
+    assert len(childlist) == DEFAULT_REPLICA_CNT
+    wait_rebuild_finish()
+
+
+@then("the request should succeed")
+def the_request_should_succeed():
+    """the request should succeed."""
+    assert pytest.exception is None
+
+
+@then("the volume expand status should be failure")
+def the_volume_expand_status_should_be_failure():
+    """the volume expand status should be failure."""
+    the_volume_resize_status_should_be_failure()
+
+
+@then("the volume is expanded to the new capacity")
+def the_volume_is_expanded_to_the_new_capacity():
+    """the volume is expanded to the new capacity."""
+    volume = ApiClient.volumes_api().get_volume(VOLUME_UUID)
+    assert volume.spec.size == VOLUME_NEW_SIZE
+
+
+@then("the volume resize status should be failure")
+def the_volume_resize_status_should_be_failure():
+    """the volume resize status should be failure."""
+    assert pytest.exception is not None
+
+
+@then("the volume should get published with expanded capacity")
+def the_volume_should_get_published_with_expanded_capacity():
+    """the volume should get published with expanded capacity."""
+    volume = ApiClient.volumes_api().get_volume(VOLUME_UUID)
+    assert hasattr(volume.spec, "target")
+    assert str(volume.spec.target.protocol) == str(Protocol("nvmf"))
+    assert volume.spec.size == VOLUME_NEW_SIZE
+
+
+@retry(wait_fixed=200, stop_max_attempt_number=30)
+def wait_pool_online(pool_id):
+    pool = ApiClient.pools_api().get_pool(pool_id)
+    assert pool.state.status == PoolStatus("Online")
+
+
+@retry(wait_fixed=500, stop_max_attempt_number=30)
+def wait_rebuild_start():
+    vol = ApiClient.volumes_api().get_volume(VOLUME_UUID)
+    childlist = vol.state.target["children"]
+    assert (
+        (len(childlist) == DEFAULT_REPLICA_CNT)
+        and (NexusState(vol.state.target["state"]) == NexusState("Degraded"))
+        and (vol.state.target["rebuilds"] == 1)
+    )
+
+
+@retry(wait_fixed=1000, stop_max_attempt_number=30)
+def wait_rebuild_finish():
+    vol = ApiClient.volumes_api().get_volume(VOLUME_UUID)
+    assert (vol.state.target["rebuilds"] == 0) and (
+        NexusState(vol.state.target["state"]) == NexusState("Online")
+    )
+
+
+@retry(wait_fixed=1000, stop_max_attempt_number=20)
+def wait_child_faulted():
+    vol = ApiClient.volumes_api().get_volume(VOLUME_UUID)
+    target = vol.state.target
+    childlist = target["children"]
+    faulted_child = None
+    for child in childlist:
+        if ChildState(child["state"]) == ChildState("Faulted"):
+            faulted_child = child["uri"]
+    assert faulted_child is not None, "Failed to find Faulted child!"
```

Comment:
Can use lambda here, can also check in case >1 are faulted

```suggestion
    faulted = list(
        filter(lambda child: child.get("state") == "Faulted", childlist)
    )
    assert len(degraded) == 1, "Failed to find 1 Faulted child"
```